storage/sealer/sched_assigner_common.go

```diff
@@ -52,6 +65,11 @@ func (a *AssignerCommon) TrySched(sh *Scheduler) {
 	// Step 1
 	throttle := make(chan struct{}, windowsLen)
 
+	partDone := metrics.Timer(sh.mctx, metrics.SchedAssignerCandidatesDuration)
+	defer func() {
+		partDone()
+	}()
```

Comment:
Can you leave a comment here saying that this defer is meant to call the latest value of `partDone` in case we error out somewhere.